pkg/services/ngalert/migration/service.go

```diff
@@ -51,7 +51,7 @@ func ProvideService(
 // If the migration status in the kvstore is set and both unified alerting is disabled and ForceMigration is set to true, the migration will be reverted.
 func (ms *MigrationService) Run(ctx context.Context) error {
 	var errMigration error
-	errLock := ms.lock.LockExecuteAndRelease(ctx, actionName, time.Minute*10, func(context.Context) {
```

Comment:
ah this seems to be it. the ctx was not passed along

Reply:
I think it does pass [here](https://github.com/grafana/grafana/blob/e4c1a7a141683604e035fa1a07fde975fde30d8b/pkg/infra/serverlock/serverlock.go#L63)

Reply:
Do the migrations still need to be moved from `Run` to `Init` then?

Reply:
This is not relevant to the fix, just small improvements I came across while figuring this out, clean as you go and all that. I can remove them and add them to the next migration-related PR, if you prefer.